src/functions-and-arrays.js

```diff
@@ -1,73 +1,504 @@
 // Iteration #1: Find the maximum
+const maxOfTwoNumbers = (nbr1, nbr2) => {
+  return nbr1 > nbr2 ? nbr1 : nbr2;
+};
 
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
+];
+const testIfInvalid = array => {
+  if (array.length === 0) {
+    return true;
+  }
+};
+const findLongestWord = arr => {
+  if (testIfInvalid(arr)) {
+    return null;
+  }
+  let longestWord = "";
+  arr.forEach(item => {
+    if (item.length > longestWord.length) {
+      longestWord = item;
+    }
+  });
+  return longestWord;
+};
 
 // Iteration #3: Calculate the sum
-
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+const sumNumbers = arr => {
+  let counter = 0;
+  for (let i = 0; i < arr.length; i += 1) {
+    counter += arr[i];
+  }
+  return counter;
+};
+
+//Use of reduce
+// const sumNumbers = arr => {
+// testIfInvalid(arr) ? 0 : arr.reduce((acc, cur) => acc + cur);
+
+// Bonus: Calculate the sum
+const mixedArr = [6, 12, "miami", 1, true, "barca", "200", "lisboa", 8, 10];
+const hasUnsuporttedDataType = arr => {
+  let result;
+  arr.forEach(element => {
+    if (typeof element === "object") {
+      result = true;
+    }
+  });
+  return result;
+};
+const sum = arr => {
+  if (hasUnsuporttedDataType(arr)) {
+    throw new Error("Unsupported data type sir or ma'am");
+  }
+  let counter = 0;
+  for (let i = 0; i < arr.length; i += 1) {
```

Comment:
I would personally have put in the "hasUnsporttedDataType" checking here:
if(typeof arr[i] == "object" ) throw new Error("Unsupported data type sir or ma'am")

Makes it a bit cleaner in my opinion, however it's just a personal preference